src/Workspaces/Core/Portable/Packaging/IPackageInstallerService.cs

```diff
@@ -28,7 +29,7 @@ internal interface IPackageInstallerService : IWorkspaceService
         bool CanShowManagePackagesDialog();
         void ShowManagePackagesDialog(string packageName);
 
-        ImmutableArray<PackageSource> GetPackageSources();
+        ValueTask<ImmutableArray<PackageSource>> GetPackageSourcesAsync(bool allowSwitchToMainThread, CancellationToken cancellationToken);
```

Comment:
maybe TryGetPackageSourcesAsync and doc this may be initialized in the bG and may not return results initially but may return results later.

Reply:
Updated